tests/contrib/BUILD

```diff
@@ -57,3 +58,53 @@ compare_ids_test(
     id = "d86477c0011ee23ba00c9d2ffce09d15a0a9282e9af1f54309ecbb7b2c6736cc",
     images = ["@distroless_fixed_id//image"],
 )
+
+create_failing_test_for_compare_ids_test(
+    name = "test_compare_ids_test_diff_ids_fails",
+    images = [
+        "@distroless_fixed_id//image",
+        "@distroless_fixed_id_2//image",
+    ],
+)
+
+create_failing_test_for_compare_ids_test(
+    name = "test_compare_ids_test_wrong_id_fails",
+    id = "wrong_id",
+    images = ["@distroless_fixed_id//image"],
+)
+
+create_failing_test_for_compare_ids_test(
+    name = "test_compare_ids_test_one_tar_no_id_fails",
+    images = ["@distroless_fixed_id//image"],
+    reg_exps = [".*One tar provided\. Need either second tar or an id to compare it to\..*"],
+)
+
+create_failing_test_for_compare_ids_test(
+    name = "test_compare_ids_test_no_images_with_id_fails",
+    id = "some_id",
+    images = [],
+    reg_exps = [".*No images provided for test.*"],
+)
+
+genrule(
+    name = "make_invalid_tar",
+    outs = ["invalid.tar"],
+    cmd = "touch random_file.txt && tar -c random_file.txt > $@",
+)
+
+create_failing_test_for_compare_ids_test(
+    name = "test_compare_ids_invalid_tar_fails",
+    id = "some_id",
+    images = ["invalid.tar"],
+    reg_exps = [".*Unable to extract manifest\.json, make sure .* is a valid docker image.*"],
+)
+
+# This test makes sure that passing tests don't pass the failure case test
+create_failing_test_for_compare_ids_test(
+    name = "passing_compare_ids_test_test",
+    images = [
+        ":duplicate1",
+        ":duplicate2",
+    ],
+    negate = True,
```

Comment:
I don't quite understand this one. iiuc, `negate` alters the check for regex. If negate is true, then we see a match, it will fail. We kinda of using negate to foo it here?

Also, make it clear in the rule documentation that:

If negate is false, then all the input regex should match.
If negate is true, none of the input regex should match.
(If I understand your logic correctly)